extensions/agroal/deployment/src/main/java/io/quarkus/agroal/deployment/AgroalProcessor.java

```diff
@@ -307,4 +315,64 @@ HealthBuildItem addHealthCheck(AgroalBuildTimeConfig agroalBuildTimeConfig) {
         return new HealthBuildItem("io.quarkus.agroal.runtime.health.DataSourceHealthCheck",
                 agroalBuildTimeConfig.healthEnabled, "datasource");
     }
+
+    // TODO: introduce more metrics
+    @BuildStep
+    void registerMetrics(AgroalRuntimeConfig agroalRuntimeConfig,
+            // Capabilities capabilities,
+            BuildProducer<MetricBuildItem> metrics) {
+        // FIXME::: here should be this
+        // if(capabilities.isCapabilityPresent(Capabilities.METRICS))
+        // this depends on pull request 4987 being merged
+        Metadata activeCountMetadata = Metadata.builder()
+                .withName("agroal.active.count")
+                .withDescription("Count of active connections")
+                .withType(MetricType.GAUGE)
+                .build();
+        Metadata acquireCountMetadata = Metadata.builder()
+                .withName("agroal.acquire.count")
+                .withDescription("Count of acquired connections")
+                .withType(MetricType.COUNTER)
+                .build();
+        if (agroalRuntimeConfig.defaultDataSource != null) {
+            /*
+             * TODO: what is the correct thing to do when metrics for the extension are enabled,
+             * but a data source has its metrics collection disabled? Should we log a warning that the values
+             * will be zero, or should we not publish metrics for this data source?
```

Comment:
But on which data sources, if there are many? On all of them? And even if the user explicitly set `enable-metrics=false` on some data source? It would probably be fine to automatically enable it if the user didn't specify `enable-metrics` at all, but overriding the user's decision doesn't seem right.
 And I'm not sure if we have the facility to find out if the user specified `false` or didn't specify it at all (in which case it defaults to `false` too)

Reply:
So we need to switch to a String boolean `enable-metrics` with a default of auto :

- true: always enabled
- false: never enabled
- auto: enabled if smallrye-metrics is in the classpath ..

Or we need to switch to an Optional config item, this should also do the trick

Reply:
With the optional config item, you mean that if it's undefined, we will treat is as "auto"? I think that would make sense. Sounds better to me than turning it into a string. 

Btw this could get affected by the incoming config changes https://github.com/quarkusio/quarkus/pull/5387, I'm particularly worried about the "Tighten rules about accessing run time configuration values from build steps.", but I'm not familiar with what exactly the changes will be

Reply:
Yes, you're right, datasource `enable-metrics` is RUNTIME so cannot be used inside a build step.
Your warning must be emitter inside your Metric classes or inside a recorder that is used inside a RUN_TIME build step.

Reply:
Hm, I just changed that to inject the runtime config into the processor and only pass it to a recorder, but it seems https://github.com/quarkusio/quarkus/issues/5514 will forbid this usage. So I will change it again.

Reply:
Injecting to the build step and passing it to the recorder is OK for now.  We don't yet support directly injecting the config into the recorder so that has to be implemented before we can take any other approach.